dali/operators/reader/numpy_reader_op.cc

```diff
@@ -223,18 +223,41 @@ void NumpyReaderCPU::RunImpl(HostWorkspace &ws) {
     if (need_transpose_[i])  // 2x if transposition is required
       task_sz *= 2;
 
-    thread_pool.AddWork([&, i](int tid) {
-      if (need_slice_[i] && need_transpose_[i]) {
+    if (need_slice_[i] && need_transpose_[i]) {
+      thread_pool.AddWork([&, i](int tid) {
         SlicePermuteHelper(output[i], file_i.data, rois_[i], fill_value_);
-      } else if (need_slice_[i]) {
+      }, task_sz);
+    } else if (need_slice_[i]) {
+      thread_pool.AddWork([&, i](int tid) {
         SliceHelper(output[i], file_i.data, rois_[i], fill_value_);
-      } else if (need_transpose_[i]) {
+      }, task_sz);
+    } else if (need_transpose_[i]) {
+      thread_pool.AddWork([&, i](int tid) {
         TransposeHelper(output[i], file_i.data);
+      }, task_sz);
+    } else {
+      constexpr int64_t kThreshold = 64000;
```

Comment:
Do you plan to paralyze other steps as well (`SlicePermuteHelper`, `SliceHelper` and `TransposeHelper`)?

Reply:
Yes (separate PR).